src/ray/raylet/local_object_manager.cc

```diff
@@ -606,14 +606,18 @@ void LocalObjectManager::RecordMetrics() const {
                                                        "Restored");
 }
 
-int64_t LocalObjectManager::GetPinnedBytes() const {
-  if (pinned_objects_size_ > 0) {
-    return pinned_objects_size_;
+int64_t LocalObjectManager::GetPrimaryBytes() const {
+  return pinned_objects_size_ + num_bytes_pending_spill_;
+}
+
+bool LocalObjectManager::HasLocallySpilledObjects() const {
+  if (is_external_storage_type_fs_) {
+    return false;
   }
   // Report non-zero usage when there are spilled / spill-pending live objects, to
   // prevent this node from being drained. Note that the value reported here is also
   // used for scheduling.
-  return (spilled_objects_url_.empty() && objects_pending_spill_.empty()) ? 0 : 1;
+  return !spilled_objects_url_.empty();
```

Comment:
Hmm longer term I think we should introduce an object store disk resource, and report these bytes separately there. It could be confusing to mix the two storages into one.

Reply:
Makes sense, I'll leave it as is then.